app/models/sensor.rb

```diff
@@ -1,16 +1,19 @@
 class Sensor < ApplicationRecord
   enum status: {active: 0, inactive: 1, error: 2}
 
-  belongs_to :zone
+  belongs_to :zone, counter_cache: true
   has_many :sensor_logs, dependent: :destroy
   has_many :alerts, dependent: :nullify
 
+  # rubocop:disable Rails/HasManyOrHasOneDependent
+  has_one :latest_log, ->{order(created_at: :desc)}, class_name: "SensorLog"
```

Comment:
```suggestion
  has_one :latest_log, ->{order(created_at: :desc)}, class_name: SensorLog.name
```
Viết ra scope order(created_at: :desc), rồi call lại cho gọn